drivers/scheduler/k8s/k8s.go

```diff
@@ -3019,6 +2993,49 @@ func (k *K8s) addLabelsToPVC(pvc *v1.PersistentVolumeClaim, labels map[string]st
 	}
 }
 
+// CreateAutopilotRule creates the AutopilotRule object
+func (k *K8s) CreateAutopilotRule(apRule apapi.AutopilotRule) (*apapi.AutopilotRule, error) {
+	t := func() (interface{}, bool, error) {
+		apRule.Labels = defaultTorpedoLabel
+		aRule, err := k8sAutopilot.CreateAutopilotRule(&apRule)
+		if errors.IsAlreadyExists(err) {
+			if rule, err := k8sAutopilot.GetAutopilotRule(apRule.Name); err == nil {
+				logrus.Infof("Using existing AutopilotRule: %v", rule.Name)
+				return aRule, false, nil
+			}
+		}
+		if err != nil {
+			return nil, true, fmt.Errorf("Failed to create autopilot rule: %v. Err: %v", apRule.Name, err)
+		}
+
+		return aRule, false, nil
+	}
+
+	apRuleObj, err := task.DoRetryWithTimeout(t, k8sObjectCreateTimeout, DefaultRetryInterval)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Infof("Created autopilot rule: %+v", apRuleObj)
+
+	return apRuleObj.(*apapi.AutopilotRule), nil
+}
+
+// UpdateAutopilotRule updates the AutopilotRule
```

Comment:
nit: no need to add function comments here. For interfaces, function comments need to be present only when we declare the interface as that provides the contract. The implementations just honor the contract.

Reply:
if I remove comment, I'm getting this error from golint:
 ./drivers/scheduler/k8s/k8s.go:3015:1: exported method K8s.UpdateAutopilotRule should have comment or be unexported